rdbi-recipes/src/main/java/com/lithium/dbi/rdbi/ratelimiter/BlockingRateLimiter.java

```diff
@@ -0,0 +1,35 @@
+package com.lithium.dbi.rdbi.ratelimiter;
+
+import com.google.common.base.Throwables;
+
+import java.util.OptionalLong;
+
+public class BlockingRateLimiter implements Limiter {
+    private final Limiter delegate;
+
+    public BlockingRateLimiter(Limiter delegate) {
+        this.delegate = delegate;
+    }
+
+    @Override
+    public boolean acquire() {
+        return !getOptionalWaitTimeForPermit().isPresent();
+    }
+
+    @Override
+    public OptionalLong getOptionalWaitTimeForPermit() {
```

Comment:
yeah i don't disagree - with the confusing-ness or the naming issues, but i was trying to be conservative re: changing the existing api. That's not a great excuse though.. i'll re-look at it and try to clarify.